pandas/tests/reshape/merge/test_merge.py

```diff
@@ -1800,3 +1801,31 @@ def test_merge_on_indexes(self, left_df, right_df, how, sort, expected):
                           how=how,
                           sort=sort)
         tm.assert_frame_equal(result, expected)
+
+class TestMergeSparseDataFrames(object):
+    # Cannot seem to get 0 or 1 working with sparse data frame
+    @pytest.mark.parametrize('fill_value,how', itertools.product([np.nan], ['left', 'right', 'outer', 'inner']))
+    def test_merge_two_sparse_frames(self, fill_value, how):
+        dense_evens = pd.DataFrame({'A': list(range(0, 200, 2)), 'B': np.random.randint(0,100, size=100)})
+        dense_threes = pd.DataFrame({'A': list(range(0, 300, 3)), 'B': np.random.randint(0,100, size=100)})
+
+        dense_merge = dense_evens.merge(dense_threes, how=how, on='A')
+
+        # If you merge two dense frames together it tends to default to float64 not the original dtype
+        dense_merge['B_x'] = dense_merge['B_x'].astype(np.int64, errors='ignore')
```

Comment:
This to me seems kind of bizarre and I couldn't find an issue for it but basically:

If you merge two dense frames together that I defined above the dtype goes from int64 to float64. I think I know where the code is that's doing that so I could fix it but didn't want to get too side tracked in this work.

Reply:
That's since `np.nan` is the missing value indicator, which is a float. Doing the merge induces missing values for `how=left/right/outer`, so the ints are cast to floats.